hive/agents/qnets/utils.py

```diff
@@ -20,9 +20,12 @@ def calculate_output_dim(net, input_shape):
     """
     if isinstance(input_shape, int):
         input_shape = (input_shape,)
-    placeholder = torch.zeros((0,) + tuple(input_shape))
+    placeholder = torch.zeros((1,) + tuple(input_shape)).to(device)
```

Comment:
Device shouldn't need to be passed in right? Like at this point it should all be on the cpu? I am fine with the change, just not sure why it's necessary

Reply:
@dapatil211 The problem is here we either need to create a dummy `hidden_state` on CPU and pass it to the network or add an if condition in the forward function of ConvRNNNetwork to take care of `hidden_state = None`. We did the latter but there might be another case other than `calculate_output_dim` where `hidden_state` is None and in that case, our `hidden_state` inside the forward function should be on the _device. (We had to change the empty array because lstm was not able to process it)

Reply:
Well don't you only need to do that because you are doing `.to(device)` here when you don't really need to?: https://github.com/chandar-lab/RLHive/pull/258/files/4874d7a579fbcaf3c26dce7d3fecc01c5c57fac6#diff-324043212c24b1c8e9c8c139a4defa49947886ee8fc9f4f9ab8138d4347f4860R139-R141

Reply:
fixed